README.md

```diff
@@ -3,7 +3,46 @@
 OpenShift LightSpeed (OLS) is an AI powered assistant that runs on OpenShift and provides answers to product questions using backend LLM services.
 
 
-## Prerequisites
+<!-- vim-markdown-toc GFM -->
+
+* [Prerequisites](#prerequisites)
+* [Installation](#installation)
+    * [1. Clone the repo](#1-clone-the-repo)
+    * [2. Install python packages](#2-install-python-packages)
+    * [3. Get API keys](#3-get-api-keys)
+    * [4. Store local copies of API keys securely](#4-store-local-copies-of-api-keys-securely)
+    * [5. Configure OpenShift LightSpeed (OLS)](#5-configure-openshift-lightspeed-ols)
+    * [6. Configure LLM providers](#6-configure-llm-providers)
+    * [7. Configure OLS Authentication](#7-configure-ols-authentication)
+    * [8. Configure OLS TLS communication](#8-configure-ols-tls-communication)
+    * [9. (Optional) Configure the local document store](#9-optional-configure-the-local-document-store)
+    * [10.  (Optional) Configure conversation cache](#10--optional-configure-conversation-cache)
+    * [11. (Optional) Incorporating additional CA(s). You have the option to include an extra TLS certificate into the OLS trust store as follows.](#11-optional-incorporating-additional-cas-you-have-the-option-to-include-an-extra-tls-certificate-into-the-ols-trust-store-as-follows)
+    * [12. Registering new LLM provider](#12-registering-new-llm-provider)
+* [Usage](#usage)
+    * [Deployments](#deployments)
+        * [Local Deployment](#local-deployment)
+            * [Run the server](#run-the-server)
+        * [Optionally run with podman](#optionally-run-with-podman)
+        * [Optionally run inside an OpenShift environment](#optionally-run-inside-an-openshift-environment)
+    * [Communication with the service](#communication-with-the-service)
+        * [Query the server](#query-the-server)
+        * [Swagger UI](#swagger-ui)
+        * [OpenAPI](#openapi)
+        * [Metrics](#metrics)
+        * [Gradio UI](#gradio-ui)
+        * [Swagger UI](#swagger-ui-1)
+    * [Deploying OLS on OpenShift](#deploying-ols-on-openshift)
+* [Project structure](#project-structure)
+    * [Overall architecture](#overall-architecture)
+    * [Sequence diagram](#sequence-diagram)
+    * [Token truncation algorithm](#token-truncation-algorithm)
+* [Contributing](#contributing)
+* [License](#license)
+
+<!-- vim-markdown-toc -->
```

Comment:
It seems these comments related to `vim-markdown-toc` are not neccessary.

Reply:
it was added by me. If these comments are there, the ToC will be automatically regenerated in the Editor. It's based on https://github.com/mzlogin/vim-markdown-toc but these comments seems to be compatible with other plugins for other IDEs as well.

Reply:
added comments about comments @raptorsun